### ISSUE TITLE
Generate code details for ave features

### ISSUE THREAD

**Issue by xiaoniuchushi** [ZiqiaoPeng/SyncTalk#53]
May I ask why 16 needs to be subtracted here. After subtracting 16, the length of the generated data self. data_len does not match the result in the may you provided.

![20240407-160350](https://github.com/ZiqiaoPeng/SyncTalk/assets/10252589/59cc2ede-9b64-4ceb-b12d-a36cbf6c7805)

https://github.com/ZiqiaoPeng/SyncTalk/blob/9ca6a0dd4bdf5c3cd4b571af31265a11e8e9e0d5/nerf_triplane/utils.py#L1623

**Comment by WoJiaoFuXiaoYun** [comment]
Same problem, add an error message

```
Namespace(H=450, O=True, W=450, amb_aud_loss=1, amb_dim=2, amb_eye_loss=1, asr=False, asr_model='ave', asr_play=False, asr_save_feats=False, asr_wav='', att=2, aud='', bg_img='', bound=1, bs_area='upper', ckpt='latest', color_space='srgb', cuda_ray=True, data_range=[0, -1], density_thresh=10, density_thresh_torso=0.01, dt_gamma=0.00390625, emb=False, exp_eye=True, fbg=False, finetune_lips=False, fix_eye=-1, fovy=21.24, fp16=True, fps=50, gui=False, head_ckpt='', ind_dim=4, ind_dim_torso=8, ind_num=20000, init_lips=False, iters=60000, l=10, lambda_amb=0.1, lr=0.01, lr_net=0.001, m=50, max_ray_batch=4096, max_spp=1, max_steps=16, min_near=0.05, num_rays=65536, num_steps=16, offset=[0, 0, 0], part=False, part2=False, patch_size=1, path='data/XiaoTie', portrait=False, preload=0, pyramid_loss=0, r=10, radius=3.35, scale=4, seed=0, smooth_eye=False, smooth_lips=False, smooth_path=False, smooth_path_window=7, test=False, test_train=False, torso=False, torso_shrink=0.8, train_camera=False, unc_loss=1, update_extra_interval=16, upsample_steps=0, warmup_step=10000, workspace='model/XiaoTie')
[INFO] load 4268 train frames.
Traceback (most recent call last):
  File "main.py", line 219, in <module>
    train_loader = NeRFDataset(opt, device=device, type='train').dataloader()
  File "/home/sg/github/SyncTalk/nerf_triplane/provider.py", line 179, in __init__
    for mel in data_loader:
  File "/home/sg/miniconda3/envs/sync/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 630, in __next__
    data = self._next_data()
  File "/home/sg/miniconda3/envs/sync/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 674, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/home/sg/miniconda3/envs/sync/lib/python3.8/site-packages/torch/utils/data/_utils/fetch.py", line 51, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/sg/miniconda3/envs/sync/lib/python3.8/site-packages/torch/utils/data/_utils/fetch.py", line 51, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/sg/github/SyncTalk/nerf_triplane/utils.py", line 1646, in __getitem__
    raise Exception('mel.shape[0] != 16')
Exception: mel.shape[0] != 16
```

**Comment by WoJiaoFuXiaoYun** [comment]
Have you solved it?

**Comment by WoJiaoFuXiaoYun** [comment]
found that it has been fixed : efb4bba9de0542053cee0d8652a512e66dc0a1df

**Comment by ZiqiaoPeng** [comment]
This bug has been fixed.
